src/base/static/libs/maps/mapboxgl-provider.js

```diff
@@ -433,21 +433,21 @@ export default (container, options) => {
     ];
   };
 
-  const createVectorTileLayer = async ({
-    id,
-    url,
-    style_url,
-    source_layer,
-  }) => {
+  const createVectorTileLayer = async ({ id, url, layers, source_layer }) => {
     sourcesCache[id] = {
       type: "vector",
       tiles: [url],
     };
     map.addSource(id, sourcesCache[id]);
 
-    const style = await VectorTileClient.fetchStyle(style_url);
+    // If layers is an array, assume we have a config-based set of mapbox
+    // layers. Otherwise, assume we have a url to a remote-hosted mapbox
+    // stylesheet.
+    layers = Array.isArray(layers)
+      ? layers
+      : await VectorTileClient.fetchLayers(layers);
 
-    layersCache[id] = style.layers.map(layer => {
+    layersCache[id] = layers.map(layer => {
       layer.source = id;
       layer["source-layer"] = source_layer;
       return layer;
```

Comment:
Not related to this PR, but we should fix this mutation of our `layer` config objects:

```javascript
layers.map(layer => ({
  ...layer,
  source: id,
  ['source-layer']: source_layer,
}))
```

Does that make sense?